src/extract-cli.js

```diff
@@ -56,21 +56,26 @@ const extractor = new extract.Extractor({
 
 files.forEach(function(filename) {
   let file = filename;
-  const ext = file.split('.').pop();
+  let ext = file.split('.').pop();
   if (ALLOWED_EXTENSIONS.indexOf(ext) === -1) {
     console.log(`[${PROGRAM_NAME}] will not extract: '${filename}' (invalid extension)`);
     return;
   }
   console.log(`[${PROGRAM_NAME}] extracting: '${filename}`);
   try {
     let data = fs.readFileSync(file, {encoding: 'utf-8'}).toString();
-    extractor.parse(file, extract.preprocessTemplate(data, ext));
 
-    if (ext !== 'js') {
-      data = extract.preprocessScriptTags(data, ext);
+    if (ext !== 'js' || ext !== 'ts') {
+      const script = extract.preprocessScriptTags(data, ext);
+      data = script.content;
+      if (script.lang) {
+        ext = script.lang;
+      }
     }
 
-    extractor.parseJavascript(file, data);
+    if (ext !== 'ts') {
+      extractor.parseJavascript(file, data);
+    }
```

Comment:
that seems incorrect to me (you should add tests for the cli !) but at the very least ``extract.preprocessTemplate`` is not called anymore. Is that intentional ?

Also, I'd rather (but it's less important)
- use a construct that enables "script file extensions" instead of a series of ``ext !== 'js'``: ['js', 'ts'].includes(ext) is better.
- try/catch exceptions when trying to parse the Javascript, instead of relying on the extension. It's way more defensive and will allow you to parse exotic extensions ^^ you would not need the second "if"

Reply:
> extract.preprocessTemplate is not called anymore. Is that intentional ?

It's removed accidentally... I reverted it at c36956d. Sorry! 
Tests for cli should be added.

Reply:
> ['js', 'ts'].includes(ext) is better.

Thank you! I'll use the code after this ^^